crates/project/src/git_store.rs

```diff
@@ -4359,14 +4361,20 @@ impl Repository {
                     "start updating index text for buffer {}",
                     path.as_unix_str()
                 );
+
+                let executable = match fs.metadata(&abs_path).await {
+                    Ok(Some(meta)) => meta.is_executable,
+                    Ok(None) => false,
+                    Err(_err) => false,
+                };
```

Comment:
When the above is addressed, we should move it inside `RepositoryState::Local` prong.